src/sentry/sentry_metrics/visibility/metrics_blocking.py

```diff
@@ -14,36 +14,63 @@ class BlockedMetricsRelayConfig(TypedDict):
     deniedNames: Sequence[str]
 
 
+@dataclass(frozen=True)
+class MetricOperation:
+    metric_mri: str
+    block_tags: Set[str]
+    unblock_tags: Set[str]
+    # This can be None, since if it is None, it implies that no state change will be applied to the metric.
+    block_metric: Optional[bool] = None
+
+    def to_blocked_metric(self) -> "BlockedMetric":
+        return BlockedMetric(
+            metric_mri=self.metric_mri,
+            is_blocked=self.block_metric is True,
+            blocked_tags=self.block_tags,
+        )
+
+
 @dataclass(frozen=True)
 class BlockedMetric:
     metric_mri: str
-    tags: Set[str]
+    is_blocked: bool
+    blocked_tags: Set[str]
 
     @classmethod
     def from_dict(cls, dictionary: Mapping[str, Any]) -> Optional["BlockedMetric"]:
         if "metric_mri" not in dictionary:
             return None
 
         return BlockedMetric(
-            metric_mri=dictionary["metric_mri"], tags=set(dictionary.get("tags") or [])
+            metric_mri=dictionary["metric_mri"],
+            is_blocked=dictionary["is_blocked"],
+            blocked_tags=set(dictionary.get("blocked_tags") or []),
         )
 
-    def merge(self, other: "BlockedMetric") -> "BlockedMetric":
-        return BlockedMetric(metric_mri=self.metric_mri, tags=self.tags.union(other.tags))
+    def apply(self, other: MetricOperation) -> "BlockedMetric":
+        return BlockedMetric(
+            metric_mri=self.metric_mri,
+            is_blocked=other.block_metric if other.block_metric is not None else self.is_blocked,
+            blocked_tags=self.blocked_tags.union(other.block_tags) - other.unblock_tags,
+        )
+
+    def is_useless(self):
```

Comment:
maybe something like: 
```suggestion
    def is_empty_rule(self):
```

Reply:
Will fix in the follow up to not slow down the deployment.